src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/hmi/get_device_connection_status_request.cc

```diff
@@ -0,0 +1,220 @@
+  const hmi_apis::Common_UserSetting::eType usb_transport_status =
+      application_manager_.GetPolicyHandler().GetDeviceUSBTransportStatus(
+          device_id);
+
+  current_device[strings::name] = device_name;
+  current_device[strings::id] = device_id;
+  current_device[strings::usb_transport_status] = usb_transport_status;
+}
+
+bool DeviceDataRetriever::GetDeviceName(const std::string& device_id,
+                                        std::string& out_device_name) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  connection_handler::DeviceHandle device_handle;
+  connection_handler::ConnectionHandler& connection_handler =
+      application_manager_.connection_handler();
+  if (!connection_handler.GetDeviceID(device_id, &device_handle)) {
+    LOG4CXX_WARN(logger_,
+                 "Failed to extract device handle for device: " << device_id);
+    return false;
+  }
+
+  const protocol_handler::SessionObserver& session_observer =
+      connection_handler.get_session_observer();
+  const int32_t invalid_result = -1;
+  if (invalid_result ==
+      session_observer.GetDataOnDeviceID(
+          device_handle, &out_device_name, NULL, NULL, NULL)) {
+    LOG4CXX_WARN(logger_,
+                 "Failed to extract device name for device: "
+                     << device_id << " with handle: " << device_handle);
+    return false;
+  }
+
+  return true;
+}
+
+GetDeviceConnectionStatusRequest::GetDeviceConnectionStatusRequest(
+    const application_manager::commands::MessageSharedPtr& message,
+    ApplicationManager& application_manager,
+    rpc_service::RPCService& rpc_service,
+    HMICapabilities& hmi_capabilities,
+    policy::PolicyHandlerInterface& policy_handler)
+    : RequestFromHMI(message,
+                     application_manager,
+                     rpc_service,
+                     hmi_capabilities,
+                     policy_handler) {}
+
+GetDeviceConnectionStatusRequest::~GetDeviceConnectionStatusRequest() {}
+
+void GetDeviceConnectionStatusRequest::Run() {
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  smart_objects::SmartObject& object = *message_;
+  smart_objects::SmartObject& msg_params_so = object[strings::msg_params];
+
+  if (msg_params_so.keyExists(strings::device) &&
+      !msg_params_so[strings::device].empty()) {
+    smart_objects::SmartObject& devices = msg_params_so[strings::device];
+    DeviceDataRetriever device_data_retriever(devices, application_manager_);
+    for (size_t i = 0; i < devices.length(); ++i) {
+      smart_objects::SmartObject& current_device = devices[i];
+
+      std::string device_id;
+      if (!GetMandatoryParameters(current_device, device_id)) {
+        LOG4CXX_WARN(logger_,
+                     "Invalid data for device: " << device_id << " found");
+        SendErrorResponse(
+            correlation_id(),
+            static_cast<hmi_apis::FunctionID::eType>(function_id()),
+            hmi_apis::Common_Result::INVALID_DATA,
+            "");
```

Comment:
@ZhdanovP could you omit the last parameter in `SendErrorResponse`?

Reply:
@AKalinich-Luxoft it should be empty string